src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1634,6 +1637,21 @@ namespace cryptonote
     return true;
   }
   //-----------------------------------------------------------------------------------------------
+  bool core::check_service_node_time()
+  {
+    //TODO sean
+    for (size_t n = 0; n < 5; ++n)
+    {
+      crypto::public_key key = m_service_node_list.get_random_pubkey();
+      m_lmq->request(
+          key,
```

Comment:
LokiMQ uses x25519 public keys for everything (which are essentially the same cryptographic curves, but in a form that is much more efficient for encrypting, while the our primary pubkeys are ed25519, which is a better form for signing).  You can get the x25519 pubkey out of the proof given the main key using:
```C++
crypto::x25519_public_key x_pkey{0};
m_service_node_list.access_proof(pubkey, [&](auto &proof) { x_pkey = proof.pubkey_x25519; });
if (x_pkey) {
  m_lmq->request(
      x_pkey,
      ...
```